docs/reference/edot-cloud-forwarder/gcp.md

```diff
@@ -0,0 +1,287 @@
+
+### Observability and data enrichment
+
+{{edot-cf}} for GCP provides detailed context about its own health and the data it processes.
+
+- Self-telemetry: You can enable the OpenTelemetry collector's internal metrics, allowing you to monitor the service's health.
+- Metadata enrichment: By enabling the `include_metadata` option, logs are automatically enriched with context from the Pub/Sub and GCS transport layers, enabling better troubleshooting and correlation:
+  - `bucket` and `object`, for logs coming from a GCS bucket.
+  - `subscription` and `message_id`.
+  - `delivery_attempt`, useful for tracking retries.
+
+### Performance
+
+:::{warning}
+This section is still under active development. This guidance is an initial recommendation and might evolve as we refine sizing with {{motlp}}.
+:::
+
+Load tests were performed to understand how to run the ECF collector reliably in production. Tests were performed on a single Cloud Run service instance (1 vCPU) processing log files up to about 8MB in size (around 6,000 logs per file).
+
+#### Recommendations
+
+- Start with one ECF instance handling up to 10 concurrent requests. If you need to handle more traffic, add more instances rather than increasing concurrency on a single instance.
+- Use at least 512MiB of memory per Cloud Run instance. In our tests at 10 concurrent requests, peak memory usage stayed below ~430MB, so 512MiB provides safe headroom for bursts.
+- If your log files are significantly larger than 8MB, or you send many more logs per request, either lower the per‑instance concurrency or allocate more memory per instance to avoid out‑of‑memory issues.
+- ECF forwards each log once. There should not be duplicate data under normal operation.
+
+## Limitations
+
+:::{dropdown} Retry behavior for permanent errors
+:open:
+The current retry logic treats all failures the same way, whether they're temporary (for example, a brief network issue) or permanent (such as an invalid log format). This means a message that can never be processed successfully will still go through all configured retries before it is finally sent to the dead‑letter topic and archived in the GCS bucket. While this improves resilience against transient failures, it can increase processing costs for messages that were never going to succeed.
+:::
+
+:::{dropdown} Memory usage for large log files
+:open:
+ECF reads each log file fully into memory before sending it on. As a result, peak memory usage grows with both file size and the number of concurrent requests. Our recommendations (1 vCPU, 512MiB, up to 10 concurrent requests) are based on internal tests with files up to about 8MB (~6,000 logs) each. If you send much larger files or significantly more logs per request, you might need to lower per‑instance concurrency or allocate more memory per instance to avoid out‑of‑memory issues.
+:::
+
+## Remove EDOT CF
```

Comment:
Maybe let's move this to be after the `quick start` section instead?

Reply:
I'd rather keep it last, since it's a bit "end of the line"